cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -942,11 +942,14 @@ private Type getTypeImpl0(Type t, int indirectionIndex) {
   or
   indirectionIndex > 0 and
   exists(Type stripped |
-    stripped = stripPointer(t.stripTopLevelSpecifiers()) and
-    // We need to avoid the case where `stripPointer(t) = t` (which can happen on
-    // iterators that specify a `value_type` that is the iterator itself). Such a type
-    // would create an infinite loop otherwise. For these cases we simply don't produce
-    // a result for `getTypeImpl`.
+    stripped = stripPointer(t) and
```

Comment:
What's the reason behind this change?

Reply:
Fixed in dd3d70134cbd069c9fa9daa7474d55d359daa487